include/envoy/access_log/access_log.h

```diff
@@ -84,5 +84,20 @@ class Formatter {
 
 typedef std::unique_ptr<Formatter> FormatterPtr;
 
+/**
+ * Interface for access log provider
+ */
+class FormatterProvider {
+public:
+  virtual ~FormatterProvider() {}
+
+  virtual std::string format(const Http::HeaderMap& request_headers,
```

Comment:
Can you add a Doxygen comment here with args/return explanation?

Reply:
Agreed
- [ ] doxygen comment on FormatterProvider

Reply:
Felt like it was redundant to call out the args, so just added an explanation and then an `@return`

Reply:
Envoy `include/envoy` APIs do have a verbose/redundant format, the best way to think of them is as official documentation that could ship without even the code (even though nobody uses them that way today, this is the style..).